### PR TITLE
Disallow changing notifications

### LINKED ISSUE #1199: Disallow changing notifications
See neo-project/neo#1707 and neo-project/neo#1708.

### PR DESCRIPTION
**Pull request by fyrchik**
Close #1199 .

### CODE REVIEW

File: pkg/vm/stackitem/item.go [hunk 1]
```diff
@@ -1019,3 +1019,60 @@ func (i *Buffer) Convert(typ Type) (Item, error) {
 func (i *Buffer) Len() int {
 	return len(i.value)
 }
+
+// DeepCopy returns new deep copy of the provided item.
+// Values of Interop and BigInteger items are not deeply copied.
```
Comment (roman-khimov) [inline pkg/vm/stackitem/item.go:1024]: Why don't we copy `BigInteger`?

File: pkg/core/interop_system.go [hunk 2]
```diff
@@ -282,7 +282,7 @@ func runtimeNotify(ic *interop.Context, v *vm.VM) error {
 	ne := state.NotificationEvent{
 		ScriptHash: v.GetCurrentScriptHash(),
 		Name:       name,
-		Item:       stackitem.NewArray(args),
+		Item:       stackitem.DeepCopy(stackitem.NewArray(args)).(*stackitem.Array),
```
Comment (roman-khimov) [inline pkg/core/interop_system.go:285]: I think we can avoid `NewArray` here, `elem.Item()` should be enough.

File: pkg/vm/stackitem/item.go (hunk 1 above)
Comment (fyrchik) [reply pkg/vm/stackitem/item.go:1024]: I thought we always copy it in out VM opcodes, to there is no way to alter it.
But after reexamination I see, that e.g. in `NEG` we modify it directly. Fixed.

File: pkg/core/interop_system.go (hunk 2 above)
Comment (fyrchik) [reply pkg/core/interop_system.go:285]: Because argument can be a struct, and we always store an array.

File: pkg/vm/stackitem/item.go (hunk 1 above)
Comment (AnnaShaleva) [inline pkg/vm/stackitem/item.go:1024]: The comment should also be fixed.

File: pkg/vm/stackitem/item.go [hunk 3]
```diff
@@ -1019,3 +1019,60 @@ func (i *Buffer) Convert(typ Type) (Item, error) {
...
+		return NewByteArray(val)
+	case *Buffer:
+		val := make([]byte, len(it.value))
+		copy(val, it.value)
+		return NewBuffer(val)
+	case *Bool:
+		return NewBool(it.value)
+	case *Interop:
+		return NewInterop(it.value)
+	default:
```
Comment (AnnaShaleva) [inline pkg/vm/stackitem/item.go:1075]: Don't we need to copy `stackitem.Pointer` also?

Comment (fyrchik) [reply pkg/vm/stackitem/item.go:1079]: Fixed.

**Pull request merged**